packages/typespec-ts/test/integration/typespec/overload/routes.tsp

```diff
@@ -0,0 +1,67 @@
+import "@typespec/rest";
+
+using TypeSpec.Http;
+
+namespace TypeScript.Test.Overalod;
+
+@doc("Resource to process.")
+model Resource {
+  id: string;
+}
+@route("/overload")
+interface OveralodOperations {
+  /**
+   * Overload with same routes
+   */
+  @route("/get")
+  getThing(@body param: string | int32): string | int32;
+
+  @overload(OveralodOperations.getThing)
+  getString(@body param: string): string;
+
+  @overload(OveralodOperations.getThing)
```

Comment:
@m-nash, did we get to an agreement on how we'll do overloads? Does this align with that?

Reply:
I believe we agreed we would use sharedRoute in the main.tsp and remove the @overload decorator from azure.core and potentially add it for client.tsp only.

@srnagar can you confirm if that is correct?

Reply:
I moved the current overload cases to client.tsp. I notice that we have an issue https://github.com/Azure/typespec-azure/issues/2738 in Project TOM but not started yet.

Reply:
> I believe we agreed we would use sharedRoute in the main.tsp and remove the @overload decorator from azure.core and potentially add it for client.tsp only.
> 
> @srnagar can you confirm if that is correct?

Yes, that is correct.